common/components/QueueItem.tsx

```diff
@@ -142,49 +61,71 @@ export async function QueueItem({
                       d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12Z"
                     />
                   </svg>
-                </div>
-              );
+                </div>)
           })}
         </button>
       );
     } else if (type === "search") {
       return (
         <button
           onclick={async () => {
-            await handleButtonClick();
-            await sleep(250);
-            const session = await getSessionWithCode(code);
-            if (!session) return;
+            const clickCode = code.length ? code : code.val
+            const session = await getSessionWithCode(clickCode);
+            // check if session is null
+            if (!session) {
+              return;
+            }
 
-            if ((session.queue.some((v) => v.id == item.id) || session.currentlyPlaying?.id == item.id)) {
+            // check if queue already contains the video
+            if (session.queue.some((v) => v.id == item.id)) {
               return;
             }
+
             session?.$.queue.val.push(item);
-            toggleLike(code, item.id);
+
+            const recommendations = await getRecommendations(session?.$.queue.val);
+            console.log(recommendations)
+            // remove all elements from the recommended queue
+            recommendations.forEach(() => {
+              session?.$.recommendedQueue.val.pop()
```

Comment:
according to the PR #63 by one of the UIX guys, this should just be:

`session?.recommendedQueue.pop()`

Also this just clears the recommendations and gets new ones, right? In that case something like `session?.recommendedQueue.length = 0;` is probably easier and faster.

Reply:
Hmm... The thing is that when I apply this to my code, it crashes and I can't run the project. I changed smth in my code, will update it soon.

Reply:
Now it works

Reply:
>> An alternative could be to save recommendations on the client side and update recommendations based on what each user added last separately. what do you think?

= I don't think it's necessary but if you and other members of the group think so - you can do it. Feel free to add / change what you want in my code and merge it to the main brunch. Unfortunately I don't have time until the next week to work on the project but I think it would be great to merge this feature to the main brunch and proceed to complete the project.